### PR TITLE
Move some things around to allow for license updating via config reload

### PR DESCRIPTION
**Pull request by mkeeler**
The bulk of this commit is moving the LeaderRoutineManager from the agent/consul package into its own package: lib/gort. It also got a renaming and its Start method now requires a context. Requiring that context required updating a whole bunch of other places in the code.

### CODE REVIEW

File: agent/consul/leader.go [hunk 1]
```diff
@@ -979,7 +979,7 @@ func (s *Server) startConfigReplication() {
 		return
 	}
 
-	s.leaderRoutineManager.Start(configReplicationRoutineName, s.configReplicator.Run)
+	s.leaderRoutineManager.Start(context.Background(), configReplicationRoutineName, s.configReplicator.Run)
```
Comment (freddygv) [inline agent/consul/leader.go:982]: How come these don't use the context from parent functions like `establishLeadership`? Should these be `context.TODO()`?

**Review by freddygv — COMMENTED** [review]
This generally looks good to me, have a couple questions.

Did you consider other names for the gort package? This abbreviation seems a bit unintuitive. What if it was called `routine`?

File: agent/consul/leader.go (hunk 1 above)
Comment (mkeeler) [reply agent/consul/leader.go:982]: I didn't look far enou

> Did you consider other names for the gort package? This abbreviation seems a bit unintuitive. What if it was called `routine`?

I started from goroutine and just shortened it. I would be happy to rename it to `routine`.

**Comment by mkeeler** [comment]
> Did you consider other names for the gort package? This abbreviation seems a bit unintuitive. What if it was called `routine`?

I started from goroutine and just shortened it. I would be happy to rename it to `routine`.

**Review by markan — APPROVED** [review]
This generally looks good; I will note that 'gort' as a name threw me as well.

File: lib/routine/gort.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+	m.logger.Debug("stopping routine", "routine", name)
+	instance.cancel()
+
+	delete(m.routines, name)
+
+	return instance
+}
+
+func (m *Manager) StopAll() {
```
Comment (freddygv) [inline lib/routine/gort.go:134]: Can you add a test for this function?

**Review by freddygv — COMMENTED** [review]
There's some stray references to gort remaining like the dir `/lib/gort`, also the filenames under `lib/routine`.

I also left one last request in line. 

Should be all all set after that.

File: lib/routine/gort.go (hunk 2 above)
Comment (mkeeler) [reply lib/routine/gort.go]: Added.

**Review by freddygv — APPROVED** [review]
:shipit:

**Review by banks — APPROVED** [review]
Nice LGTM too! 👍 on `routine` 😄

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/374007.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit da31e0449ea6323956f8ebc942a6d76332c52e27 onto `release/1.10.x` succeeded!

**Pull request merged**